src/pykeen/triples/splitting.py

```diff
@@ -89,7 +89,10 @@ def _get_cover_deterministic(triples: MappedTriples) -> torch.BoolTensor:
     :return: shape: (n,)
         A boolean mask indicating whether the triple is part of the cover.
     """
-    df = pandas.DataFrame(data=triples.numpy(), columns=COLUMN_LABELS).reset_index()
+    if triples.shape[1] == 3:   
+        df = pandas.DataFrame(data=triples.numpy(), columns=COLUMN_LABELS[:3]).reset_index()
+    else:
+        df = pandas.DataFrame(data=triples.numpy(), columns=COLUMN_LABELS).reset_index()
```

Comment:
```suggestion
    df = pandas.DataFrame(data=triples.numpy(), columns=COLUMN_LABELS[:triples.shape[1]]).reset_index()
```

Reply:
Would this code need to be adjusted too?

The splitting procedure currently ensures that all entities already occur in training (as necessary for a transductive setting).

What about the timestamps? When we train embeddings for and do not use any parametric format, I guess we also need to ensure that all timestamps are covered in the training set (as otherwise the non-training timestamps' embeddings will not get any meaningful training signal).